subsys/bluetooth/controller/ll_sw/ull_master.c

```diff
@@ -303,30 +314,60 @@ uint8_t ll_create_connection(uint16_t scan_interval, uint16_t scan_window,
 	conn->tx_head = conn->tx_ctrl = conn->tx_ctrl_last =
 	conn->tx_data = conn->tx_data_last = 0;
 
-#if defined(CONFIG_BT_CTLR_PHY)
-	ready_delay_us = lll_radio_tx_ready_delay_get(conn_lll->phy_tx,
-						      conn_lll->phy_flags);
-#else
-	ready_delay_us = lll_radio_tx_ready_delay_get(0, 0);
-#endif
-
 	/* TODO: active_to_start feature port */
 	conn->ull.ticks_active_to_start = 0U;
 	conn->ull.ticks_prepare_to_start =
 		HAL_TICKER_US_TO_TICKS(EVENT_OVERHEAD_XTAL_US);
 	conn->ull.ticks_preempt_to_start =
 		HAL_TICKER_US_TO_TICKS(EVENT_OVERHEAD_PREEMPT_MIN_US);
-	conn->ull.ticks_slot =
-		HAL_TICKER_US_TO_TICKS(EVENT_OVERHEAD_START_US +
-				       ready_delay_us +
-				       328 + EVENT_IFS_US + 328);
 
 	lll->conn = conn_lll;
 
 	ull_hdr_init(&conn->ull);
 	lll_hdr_init(&conn->lll, conn);
 
 conn_is_valid:
+#if defined(CONFIG_BT_CTLR_PHY)
+	ready_delay_us = lll_radio_tx_ready_delay_get(conn_lll->phy_tx,
+						      conn_lll->phy_flags);
+#else
+	ready_delay_us = lll_radio_tx_ready_delay_get(0, 0);
+#endif
+
+#if defined(CONFIG_BT_CTLR_DATA_LENGTH)
+#if defined(CONFIG_BT_CTLR_PHY)
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+	conn_lll->max_tx_time = MAX(conn_lll->max_tx_time,
+				    PKT_US(PDU_DC_PAYLOAD_SIZE_MIN,
+					   lll->phy));
+	conn_lll->max_rx_time = MAX(conn_lll->max_rx_time,
+				    PKT_US(PDU_DC_PAYLOAD_SIZE_MIN,
+					   lll->phy));
+#endif /* CONFIG_BT_CTLR_ADV_EXT */
+	max_tx_time = conn_lll->max_tx_time;
+	max_rx_time = conn_lll->max_rx_time;
```

Comment:
Can you elaborate on why we keep both a max time as a local variable, and as a field in the struct?

Reply:
the fields in the structure are only available  when PHY update is supported in the controller. The local variable is accordingly assigned based on conditional compilations and used down below to get the time reservation in ticks.